fdbclient/ReadYourWrites.actor.cpp

```diff
@@ -1410,12 +1417,16 @@ void ReadYourWritesTransaction::writeRangeToNativeTransaction( KeyRangeRef const
 			clearBegin = std::max(ExtStringRef(keys.begin), it.beginKey());
 			inClearRange = true;
 		} else if( !it.is_cleared_range() && inClearRange ) {
-			tr.clear( KeyRangeRef( clearBegin.toArenaOrRef(arena), it.beginKey().toArenaOrRef(arena) ), false );
+			approximateSize += updateSize ? clearBegin.toArenaOrRef(arena).expectedSize() +
+			                                    it.beginKey().toArenaOrRef(arena).expectedSize() + sizeof(KeyRangeRef)
+			                              : 0;
+			tr.clear(KeyRangeRef(clearBegin.toArenaOrRef(arena), it.beginKey().toArenaOrRef(arena)), false);
```

Comment:
There is some copying involved here, so it's probably better to create the `KeyRangeRef` once and use it to both update the size and then pass to `clear`.

Reply:
Actually, I think this may not be relevant because we won't be adding to `approximateSize` in this function.

Reply:
Removed counting here.